scripts/build_mod_jsteh.py

```diff
@@ -0,0 +1,181 @@
+"""
+build_mod_jsteh.py
+Copyright 2016 Adam Greig
+Licensed under the MIT licence, see LICENSE file for details.
+
+Generate footprints for JST-EH connectors.
+"""
+
+from __future__ import print_function, division
+
+# Settings ====================================================================
+
+# Courtyard clearance
+# Use 0.25 for IPC nominal and 0.10 for IPC least
+ctyd_gap = 0.25
+
+# Courtyard grid
+ctyd_grid = 0.05
+
+# Courtyard line width
+ctyd_width = 0.01
+
+# Silk line width
+silk_width = 0.15
+
+# Fab layer line width
+fab_width = 0.01
+
+# Ref/Val font size (width x height)
+font_size = (1.0, 1.0)
+
+# Ref/Val font thickness
+font_thickness = 0.15
+
+# Ref/Val font spacing from centre to top/bottom edge
+font_halfheight = 0.7
+
+
+# End Settings ================================================================
+
+
+import os
+import sys
+import time
+import math
+
+from sexp import parse as sexp_parse, generate as sexp_generate
+from kicad_mod import fp_line, fp_text, pad, draw_square
+
+
+def side_pth_refs(name):
+    out = []
+    ctyd_h = 8.2 + 2*ctyd_gap
+    y = ctyd_h / 2.0 + font_halfheight
+    out.append(fp_text("reference", "REF**", (0, -y-2.35),
+               "F.Fab", font_size, font_thickness))
+    out.append(fp_text("value", name, (0, y-2.35),
+               "F.Fab", font_size, font_thickness))
+    return out
+
+
+def pth_pads(pins):
+    x = (pins - 1)*1.25
+    pads = []
+    for pin in range(pins):
+        pads.append(pad(pin+1, "thru_hole", "circle", (x, 0), [1.6, 1.6],
+                        ["*.Cu", "*.Mask"], drill=0.8))
+        x -= 2.5
+    return pads
+
+
+def side_pth_silk(pins):
+    out = []
+    w = silk_width
+    centre = (0, -2.35)
```

Comment:
this centre position (reused elsewhere in fab/crtyd too) gives the top edge 6.45mm away from the pins where I think it should be 6.7 from the datasheet